metricbeat/module/docker/network/helper.go

```diff
@@ -64,18 +75,129 @@ type NetStats struct {
 	TxDropped     float64
 	TxErrors      float64
 	TxPackets     float64
+	Netstat       *sysinfotypes.NetworkCountersInfo
 	Total         *types.NetworkStats
 }
 
-func (n *NetService) getNetworkStatsPerContainer(rawStats []docker.Stat, dedot bool) []NetStats {
+// NoSumStats lists "stats", often config/state values, that can't be safely summed across PIDs
+var NoSumStats = []string{
+	"RtoAlgorithm",
+	"RtoMin",
+	"RtoMax",
+	"MaxConn",
+	"Forwarding",
+	"DefaultTTL",
+}
+
+func (n *NetService) getNetworkStatsPerContainer(client *client.Client, timeout time.Duration, rawStats []docker.Stat, cfg Config) ([]NetStats, error) {
 	formattedStats := []NetStats{}
 	for _, myStats := range rawStats {
+
+		var stats *sysinfotypes.NetworkCountersInfo
+		var err error
+		if cfg.NetworkSummary {
+			stats, err = fetchContainerNetStats(client, timeout, myStats.Container.ID)
+			if err != nil {
+				return nil, errors.Wrap(err, "error fetching per-PID stats")
+			}
+		}
+
 		for nameInterface, rawnNetStats := range myStats.Stats.Networks {
-			formattedStats = append(formattedStats, n.getNetworkStats(nameInterface, rawnNetStats, myStats, dedot))
+			singleStat := n.getNetworkStats(nameInterface, rawnNetStats, myStats, cfg.DeDot)
+			if cfg.NetworkSummary {
+				singleStat.Netstat = stats
+			}
+			formattedStats = append(formattedStats, singleStat)
 		}
 	}
 
-	return formattedStats
+	return formattedStats, nil
+}
+
+// fetchContainerNetStats gathers the PIDs associated with a container, and then uses go-sysinfo to grab the /proc/[pid]/net counters and sum them across PIDs.
+func fetchContainerNetStats(client *client.Client, timeout time.Duration, container string) (*sysinfotypes.NetworkCountersInfo, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	body, err := client.ContainerTop(ctx, container, []string{})
+	if err != nil {
+		return &sysinfotypes.NetworkCountersInfo{}, errors.Wrap(err, "error fetching container pids")
+	}
+
+	var pidPos int
+	for pos := range body.Titles {
+		if body.Titles[pos] == "PID" {
+			pidPos = pos
+			break
+		}
+	}
+
+	summedMetrics := &sysinfotypes.NetworkCountersInfo{}
+	for _, pid := range body.Processes {
+		strPID := pid[pidPos]
+		intPID, err := strconv.Atoi(strPID)
+		if err != nil {
+			return &sysinfotypes.NetworkCountersInfo{}, errors.Wrap(err, "error converting PID to int")
```

Comment:
Should we return here when failed to convert PID to int or continue to process the rest of the processes? Same question for line 150.